### ISSUE TITLE
Unable to "save" reportit template anymore as result of fix to issue #40

### ISSUE THREAD

**Issue by ikorzha** [Cacti/plugin_reportit#45]
I am sorry that have to bring another issue related to reportit.
The functionality did use to work until recent changes.
Issue is as following: Console->Reports->Select a previously locked report to edit->remove locked status and hit save..-> Results in below msg and no unlock:

(/plugins/reportit/templates.php:  65 form_save)(/plugins/reportit/templates.php: 603  input_validate_input_number)(/lib/html_validate.php: 33  die_html_input_error)(/lib/html_validate.php: 64 cacti_debug_backtrace)
--
From Cacti log: 
![image](https://user-images.githubusercontent.com/11211789/45921375-54a62280-be82-11e8-89a8-3bfdeec9d7ad.png)

**Comment by netniV** [comment]
I'll see if you have a report in your db... then I'll try and unlock it.

**Comment by ikorzha** [comment]
Don't try, there are no templates in that DB. I just started looking into this plugin seriously yesterday.
However if you can fix ;) the template import and export functionality that doesn't work I will be happy to share a report template with you from my current installation :)

**Comment by ikorzha** [comment]
Actually the issue extends further, I can't even create a new template, get similar error when I try to save it. Doesn't matter if it locked or unlocked.

**Comment by ikorzha** [comment]
Additional information: I believe this "template save problem" has been introduced when you resolved issue #40
I have just reverted changes to revision 86 as of 9/6/18 and I am able to create and save template/remove lock.
Please examine changes that were introduced in by fix in #40 I have a strong belief it is a root cause of current issue.

**Comment by netniV** [comment]
Do you have any measurands on the report template?

**Comment by ikorzha** [comment]
Yes I do have measurands on template, however, issue is even simpler prior to adding measurands.
Try creating a new reportit template, doesn't matter which data source you use and hit save, you will get error msg on attempt to save.

**Comment by netniV** [comment]
Is that with or without #41 ?

**Comment by ikorzha** [comment]
I just did an additional testing and have exact results, issue was introduced by fix in issue #40 instead of 41, (my apologies for confusion and guess work.)
Without #40  everything is working correctly, issue is introduced when update file: /reportit/include/global_forms.php to the latest revision, revision history show that last time this file was updated to resolve issue #40 
Now you have complete info.

**Comment by ikorzha** [comment]
FYI: Change that broke the Camal's back ;)
![image](https://user-images.githubusercontent.com/11211789/45923254-09563900-beaf-11e8-8961-a2af3d88c68b.png)

**Comment by netniV** [comment]
Yeah, I had already identified that.  I think the reason is that as a label, it doesn't get resubmitted when the form does so it loses the value.  There are two choices, change the type whilst still keeping it readable, or include the value as a hidden field so it's transmitted.  I'm tempted to go with the latter.

**Comment by netniV** [comment]
Should now be resolved.

**Comment by ikorzha** [comment]
Thank you I have validated the new fix, this portion of reportit is now fully functional.

I have generated my first report and I do see lots of issues, from visual to technical (unrelated to this one:) ). I will be continue opening issue 1 at a time so this plugin can get to quality standard of cacti itself
Thank you once again for working with me on it. New issue #46 is open for the next issue on the list.
